File: backend/src/domain/entities/SessionFeedback.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('session_feedbacks')
export class SessionFeedback {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'session_id' })
  sessionId: number;

  @Column({ name: 'mentee_member_id' })
  menteeMemberId: number;

  @Column({ type: 'integer', nullable: true })
  rating: number;

  @Column({ type: 'text', nullable: true })
  comment: string;

  @Column({ type: 'boolean', name: 'is_public', nullable: true })
  isPublic: boolean;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend/src/main/java/com/service/backend/mentorship/domain/entity/SessionFeedback.java
package com.service.backend.mentorship.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("session_feedbacks")
public class SessionFeedback {

    @Id
    private Integer id;

    @Column("session_id")
    private Integer sessionId;

    @Column("mentee_member_id")
    private Integer menteeMemberId;

    private Integer rating;

    private String comment;

    @Column("is_public")
    @Builder.Default
    private Boolean isPublic = true;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
